azurecloudify/server.py

```diff
@@ -167,6 +171,54 @@ def set_data_disks(azure_config, **kwargs):
     utils.write_target_runtime_properties_to_file(required_keys=None, prefixed_keys=[constants.DATA_DISK_SIZE_KEY, constants.DATA_DISK_KEY, constants.DATA_DISK_LUN_KEY],  need_suffix=None)
 
 
+def execute_vm_extensions():
+    '''Executes a Virtual Machine Extension (PowerShell scripts)'''
+    resource_group_name = \
+        ctx.instance.runtime_properties[constants.RESOURCE_GROUP_KEY]
+    vm_name = ctx.instance.runtime_properties[constants.VM_KEY]
+    headers, location, subscription_id = auth.get_credentials()
+    # Build Virtual Machine Extensions URL
+    vmx_url = '{0}/{1}/{2}/{3}/{4}/{5}?{6}'.format(
+        constants.azure_url,
+        'subscriptions/{0}'.format(subscription_id),
+        'resourceGroups/{0}'.format(resource_group_name),
+        'providers/Microsoft.Compute',
+        'virtualMachines/{0}'.format(vm_name),
+        'extensions/{0}'.format(vm_name),
+        'validating=true&api-version=2016-03-30'
+    )
+    # Configure VMX entry point and script URLs
+    ps_urls = ctx.node.properties.get('vmx_script_urls', list())
+    ps_exec = ctx.node.properties.get('vmx_script_entry')
+    # If a VMX is defined, build it
+    if ps_exec:
+        command_to_exec = 'powershell -ExecutionPolicy Unrestricted ' \
+                          '-file {0}'.format(ps_exec)
```

Comment:
This approach is on purpose.  There is no user-data / cloud-init style option for Azure that works with Windows (or, to be more precise, anything besides Ubuntu Linux).  Post-deployment changes can only be done via Virtual Machine Extensions (what I've chosen to do) or "Unattend content" which is about as sub-optimal of a solution as could possibly have been made.